2022/incoming/naughty-or-nice-networks.pod

```diff
@@ -0,0 +1,95 @@
+Author: Pete Houston <HOUSTON@cpan.org>
+Title: Naughty or Nice Networks
+Topic: Combining CIDRs into lookup lists
+
+=encoding utf8
+
+=head1 Naughty or Nice Networks
+
+The wise old elf was taking a breather from all the busy preparations
+and was just looking forward to a richly deserved hot toddy when he
+heard a familiar knock at the door. He allowed himself the briefest of
+sighs.
+
+"Come in, Wispy!"
+
+In scuttled Wispy Crinkletops wearing a worried expression.
+
+"You know the wonderful new web portal we set up so that people could
+write to us directly with their wishes? Well, not all the messages
+received have been about that. Lots are from people looking to sell us
+things - and mostly useless things at that! And there are some others
+which are just so ... rude. How could anyone be so mean?"
+
+"Now, now. Don't get all flustered. You know that this sort of thing is
+why the boss keeps a Naughty List as well as a Nice List.". The wise old
+elf thought for a moment and then asked, "Did you look to see if these
+all came from the same IP address?".
+
+Wispy composed herself and then nodded. "I did check but they were from
+quite a big number of different addresses so it would take forever to
+try to deal with them all.".
+
+"Hmmm. Can you show me the list?"
+
+Again Wispy nodded and produced a sizeable sheaf of 11 by 14 fanfold.
+The wise old elf smiled kindly and ran his eyes down the list. Some
+definite patterns were detectable and he hit upon an idea.
+
+"Even though there are a lot of different addresses, it looks like only
+a dozen of so netblocks are involved. We can use L<Net::CIDR::Lite> to
+filter those.". He turned to his terminal and quickly constructed some
+code. Wispy peered over his shoulder to see.
+
+=begin perl
+
+use CGI::Fast;
+use Net::CIDR::Lite;
+
+# Build list of netblocks
+my $naughty = Net::CIDR::Lite->new;
+$naughty->add ($_) for qw{10.52.192.0/18 172.26.64.0/20 192.168.188.0/23};
+
+while (my $fcgi = CGI::Fast->new) {
+
+    # Test for presence in the list
+    if ($naughty->find ($ENV{REMOTE_ADDR})) {
+        triage ($fcgi);
```

Comment:
Could we maybe add a stub for `sub triage` somewhere in the code so that it's clear that this isn't something that is exported by `CGI::Fast` or `Net::CIDR::Lite`?

Reply:
Sure. What about `process_form` as well?

Reply:
> Sure. What about `process_form` as well?

Good catch. I think that would be helpful as well.